### PR TITLE
[FEATURE] Create pipeLog.md

### PR DESCRIPTION
**Pull request by 3dos**
## Description
I added a new snippet

## What does your PR belong to?
- [ ] Website
- [x] Snippets
- [ ] General / Things regarding the repository (like CI Integration)
- [ ] Tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking improvement of a snippet)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have checked that the changes are working properly
- [x] I have checked that there isn't any PR doing the same
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by flxwu** [comment]
Thanks for the PR!

TBH, I don't really see the value in writing in in one statement instead of first logging and then returning the val. Opinions? @Chalarangelo @fejes713

**Comment by 3dos** [comment]
Hi, thanks for the comment. It's a little helper function I use in development while using Ramda's `pipe` or `compose` without breaking the pipeline so I thought it would be useful among the other snippets.

Obviously feel free to reject this PR :)

**Comment by flxwu** [comment]
right for debugging purposes it indeed would be quite convenient 👍

**Comment by 3dos** [comment]
Thanks for the update in the description !

File: snippets/pipeLog.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+### pipeLog
+
+Use `console.log` in a pipeline as this function simply encloses it and returns the passed value. This is especially useful for debugging when you want to log a variable's value before its usage.
+
+Logs a value and returns it.
+
+```js
+const pipeLog = data => console.log(data) || data
```
Comment (Chalarangelo) [inline snippets/pipeLog.md]: Semistandard dictates adding a semicolon (`;`) at the end of the statement.

Comment (Chalarangelo) [reply snippets/pipeLog.md]: I am also not entirely certain that this is the only way to achieve this. I think something along the lines of `console.log(data), data` would have the same effect.

File: snippets/pipeLog.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+### pipeLog
+
+Use `console.log` in a pipeline as this function simply encloses it and returns the passed value. This is especially useful for debugging when you want to log a variable's value before its usage.
+
+Logs a value and returns it.
```
Comment (Chalarangelo) [inline snippets/pipeLog.md]: This should be the first part of the description of the snippet, detailing the functionality.

File: snippets/pipeLog.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+### pipeLog
+
+Use `console.log` in a pipeline as this function simply encloses it and returns the passed value. This is especially useful for debugging when you want to log a variable's value before its usage.
```
Comment (Chalarangelo) [inline snippets/pipeLog.md]: This should be expanded to explain the how of this snippets inner workings and be less about use-cases and more about the details of how the code works (although there is very little to see here anyways). It should also come after the functionality description.

File: snippets/pipeLog.md (hunk 1 above)
Comment (skatcat31) [reply snippets/pipeLog.md]: as stated above, a comma may actually be the preferred way to handle this.

**Comment by Chalarangelo** [comment]
@skatcat31 `console.debug` could be a better option actually.

File: snippets/pipeLog.md (hunk 1 above)
Comment (3dos) [reply snippets/pipeLog.md]: I didn't know about the comma operator (every time, it's a good time to learn). Now you mention it, it totally makes sense. But I get an `Unexpected use of comma operator` from es-lint while using

```js
const debug = obj => (console.log(obj), obj);
```

I can't find any help on the [`no-sequence`](https://eslint.org/docs/rules/no-sequences) page of es-lint docs.

**Comment by skatcat31** [comment]
No, it couldn't. `console.verb` is a pattern of logging. They are usually semantic and have some differences, but in the end most do the same thing: write to standard out. It does not allow stepping, it does not allow reassignment or modification, it does not allow the true power of debugging. Log debugging is NOT a good way to debug. With powerful tools built into every browser and NodeJS itself with a simple `debugger` they aren't comparable, especially if I can't step out of a function to see its calling context. That's more often than not what really matters. I'd be hard pressed to ever justify a `console` statement, and in fact most developers are. It's often why lint rules have it set at `error`.

Still that's besides the point. I was not against this, I'm just saying it's not necessarily something I'd put in the main repo do to the intended use of the snippet and thus I'm not the best judge of it. It's helpful for people who don't know of the `debugger` statement, but it should be noted that `console` logging to inspect something is considered a bad practice and that you should use a debugger instead.

As a point in it's favor however I was going back through the Ramda gitter and noticed about 3 or 4 questions about how to do just this. So take it as you will.

**Comment by Chalarangelo** [comment]
@skatcat31 It's a code smell definitely. I think this belongs in the archive more than anything else.

**Comment by 3dos** [comment]
@skatcat31 thank you for your insight. I'll definitely take a look at the `debugger`. Yet another thing I learned today \o/

File: snippets/pipeLog.md (hunk 1 above)
Comment (flxwu) [reply snippets/pipeLog.md]: Did you read through the whole docs page? 😉 ``
Rule Details

This rule forbids the use of the comma operator, with the following exceptions:

    In the initialization or update portions of a for statement.
    If the expression sequence is explicitly wrapped in parentheses.

**Comment by Chalarangelo** [comment]
Merging with intent to move to the archive.

**Pull request merged**